src/main/java/hw8/data/MetalsColorsFormDataProvider.java

```diff
@@ -0,0 +1,38 @@
+package hw8.data;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.reflect.TypeToken;
+import hw8.entities.MetalsColorsInput;
+import org.testng.annotations.DataProvider;
+
+import java.io.FileReader;
+import java.io.IOException;
+import java.lang.reflect.Type;
+import java.util.List;
+
+public class MetalsColorsFormDataProvider {
+    @DataProvider(name = "Json Input for Metals&Colors Form")
+    public Object[][] metalsColorsFormInput() {
```

Comment:
You can just return one dimension array.

Reply:
As I understand, I can't. According to TestNG documentation, DataProvider must return array of array of Objects.
https://testng.org/doc/documentation-main.html#parameters-dataproviders

Otherwise, if I return 
`{
{inputList};
}` instead of `{ { inputList.get(0) }, { inputList.get(1) }, ... }`
the test will run only one time and I'll have to iterate the "inputList" inside the test